dali/operators/util/reshape.cc

```diff
@@ -109,21 +160,27 @@ void Reshape<Backend>::ShapeFromInput(
       }
 
       for (int d = 0; d < sample_dim; d++) {
-        output_shape_.tensor_shape_span(i)[d] = shape.tensor_data(i)[d];
+        Extent e = shape.tensor_data(i)[d];
+        int out_e = e < 0 ? -1 : relative ? round_int(e * input_shape_.tensor_shape_span(i)[d]) : e;
+        output_shape_.tensor_shape_span(i)[d] = out_e;
       }
     }
   }
 }
 
 template <typename Backend>
 template <typename TensorListLike>
-void Reshape<Backend>::ShapeFromInput(const TensorListLike &tl) {
-  TYPE_SWITCH(tl.type().id(), type2id, type,
-    (int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t),
-    (this->ShapeFromInput(view<const type>(tl));),
-    (DALI_FAIL("Reshape: shape input must have integral type; got: " + tl.type().name() +
-               " (id = " + to_string(static_cast<int>(tl.type().id())) + ")");)
-  );  // NOLINT
+void Reshape<Backend>::ShapeFromInput(const TensorListLike &tl, bool relative) {
+  if (relative) {
+    this->ShapeFromInput(view<const float>(tl));
+  } else {
+    TYPE_SWITCH(tl.type().id(), type2id, type,
+      (int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t, uint64_t),
+      (this->ShapeFromInput(view<const type>(tl));),
+      (DALI_FAIL("Reshape: shape input must have integral type; got: " + tl.type().name() +
```

Comment:
`must have integral type` unless it's the if above and we handle `float`.

Reply:
Shape input is only for absolute shape. Relative shape is specified as an argument input.
The regular input is only here for historical reasons (from before CPU/support unification).